Remove usage of tea module

baurmatt opened the pull request:
This commit remove the usage of the tea module which was used for data
type validation only. Most of the data types are already available with
the stdlib module.

Fixes #894

<!-- The following list of checkboxes are the prerequisites for getting your contribution accepted. Please check them as they are completed. -->

Pull request acceptance prerequisites:

- [x] Signed the [CLA](https://www.elastic.co/contributor-agreement/) (if not already signed)
- [x] Rebased/up-to-date with base branch
- [x] Tests pass
- [ ] Updated CHANGELOG.md with patch notes (if necessary)
- [x] Updated CONTRIBUTORS (if attribution is requested)

tylerjl commented:
Also it looks like the CI failures were legitimate; there's a few spots in which types need to be wrapped in `Optional` (if you want to skip Travis, you can check locally with `bundle exec rake intake`)

baurmatt commented:
Thanks for reviewing! :)

- I've updated to minimum required version of stdlib to 4.13.0 which is the first version those data types showed up.
- I've removed tea from the .fixtures.yml (I guess you've meant tea instead of stdlib ;))
- I've fixed the tests

Regarding the CHANGELOG: It would be great if you could do it as I'm not 100% sure which version/data/issue type I'm supposed to use.
